doc/requirements.txt

```diff
@@ -1,3 +1,3 @@
-sphinx
-sphinx-autodoc-typehints
-furo
+sphinx==7.3.7
+sphinx-autodoc-typehints==2.2.2
+furo==2024.5.6
```

Comment:
Why are you pinning the versions here?  At the least, I would expect to list only minimum, but not specific versions.

Reply:
Hm, yes this may be a tad too specific. Using `~=major.minor` or `~=major` would be a better option.

Reply:
I'd suggest leaving them unspecified or give only a minimum. What do we gain by forbidding newer versions?  It seems to have worked all this time.

Reply:
I prefer to ping dependencies at some level, often at the minor version (assuming SemVer). Having a fairly reproducible test environment is a nice thing.